COVIDseq results to Labware 8

OlintoYale opened the pull request:
First commit: This script, covid_files.R, is an R script related to the covid files: nags/summary/NGS-tracking/fastas2jason.

OlintoYale committed:
First commit: adding a new script to the repository

erinyoung commented on covid_files.R:
```diff
@@ -0,0 +1,205 @@
+
+# Summary file
+# ======================
+clade = select(read.table(file = paste(runPath,"CecretPangolin/nextclade/nextclade.tsv",
+                                       sep ="/"), sep = '\t', header = T),seqName,clade); 
+clade<-clade %>% mutate_at("seqName", str_replace, var, "")
```
It doesn't look like the nextclade clade is used anywhere. Am I missing something?

erinyoung commented on covid_files.R:
```diff
@@ -0,0 +1,205 @@
+  write.csv(select(summary,"SampleNumber","Sample_ID","collectionDate","Run Name","lineage","num_actg","clade")
+            ,dx,row.names=FALSE)
+  
+  # json file: fastas2json  
+  # ========================
+  path2python<-paste("python3 /home/Bioinformatics/Dripping_Rock/bin/fasta_to_json.py",args[1],sep = " ")
+  path2python
+  system(path2python)
```
Is there any plan to replace this with something else?

OlintoYale replied:
I have a version in R that can be added to this script but the version in python is working fine.

OlintoYale commented:
lane 139; clade is added to sumary1: summary1<-merge(summary1,clade,
by.x="df1", by.y= "seqName", all.x=TRUE)

On Fri, Jul 28, 2023 at 1:58 PM Young ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In covid_files.R
> <https://github.com/UPHL-BioNGS/General_LW_scripts/pull/32#discussion_r1278033557>
> :
>
> > +  # json file: fastas2json
> +  # ========================
> +  path2python<-paste("python3 /home/Bioinformatics/Dripping_Rock/bin/fasta_to_json.py",args[1],sep = " ")
> +  path2python
> +  system(path2python)
>
> Is there any plan to replace this with something else?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/UPHL-BioNGS/General_LW_scripts/pull/32#pullrequestreview-1552879329>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUISDYKETAN6X6GLJKNUVPDXSQKX5ANCNFSM6AAAAAA2ZH4IDM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

-- 
*************** IMPORTANT MESSAGE ***************
This message, including 
any attachments, may contain confidential information intended for a 
specific individual and purpose, and is protected by law.  If you are not 
the intended recipient, delete this message, including from trash, and 
notify me by telephone or email.


If you are not the intended recipient, 
any distributions or copying of this message, or the taking of any action 
based on its content is strictly prohibited.

OlintoYale commented:
+  # json file: fastas2json
+  # ========================
+  path2python<-paste("python3
/home/Bioinformatics/Dripping_Rock/bin/fasta_to_json.py",args[1],sep =
" ")
+  path2python
+  system(path2python)

What are you suggesting?

That is working fine, I wrote a version in R that can be added to the
covid_files script.


On Fri, Jul 28, 2023 at 2:29 PM Olinto Linares-Perdomo (DHHS) <
***@***.***> wrote:

> lane 139; clade is added to sumary1: summary1<-merge(summary1,clade,
> by.x="df1", by.y= "seqName", all.x=TRUE)
>
> On Fri, Jul 28, 2023 at 1:58 PM Young ***@***.***> wrote:
>
>> ***@***.**** commented on this pull request.
>> ------------------------------
>>
>> In covid_files.R
>> <https://github.com/UPHL-BioNGS/General_LW_scripts/pull/32#discussion_r1278033557>
>> :
>>
>> > +  # json file: fastas2json
>> +  # ========================
>> +  path2python<-paste("python3 /home/Bioinformatics/Dripping_Rock/bin/fasta_to_json.py",args[1],sep = " ")
>> +  path2python
>> +  system(path2python)
>>
>> Is there any plan to replace this with something else?
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/UPHL-BioNGS/General_LW_scripts/pull/32#pullrequestreview-1552879329>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AUISDYKETAN6X6GLJKNUVPDXSQKX5ANCNFSM6AAAAAA2ZH4IDM>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>

-- 
*************** IMPORTANT MESSAGE ***************
This message, including 
any attachments, may contain confidential information intended for a 
specific individual and purpose, and is protected by law.  If you are not 
the intended recipient, delete this message, including from trash, and 
notify me by telephone or email.


If you are not the intended recipient, 
any distributions or copying of this message, or the taking of any action 
based on its content is strictly prohibited.

OlintoYale committed:
Update covid_files.R

Added a few lines at the end to generate a file (NGS_COV_SEQ.txt) to be added at labware 8.

erinyoung commented on covid_files.R:
```diff
@@ -0,0 +1,212 @@
+  system(path2python)
+
+# File to be  added to labware 8: NGS_COV_SEQ.txt
+  # -----------------
+  path<-paste(d,"NGS_COV_SEQ.txt", sep ="/")
+  f1 <-select(summary,1,13,2,3,8)
+  col_names<-c("LIMS_TEST_ID","Run Name","UPHL Submission ID","Lineage","Non-Ambiguous Bases"); colnames(f1)<-col_names;
+  write.table(f1, path, sep =",", col.names = T, row.names = F, quote = F)
+
```
This looks like it will work perfect!

erinyoung commented:
I think this looks great! Can you update README.md with how to use this script?

OlintoYale committed:
Second Update on covid_files.R

1. Added two more libraries: "gargle","googlesheets4"

2. Added a few lines at the end that are used to send the run/results to the SARS-Cov-2 NGS Tracking Finished tab

OlintoYale committed:
Update covid_files.R

Minor modification

erinyoung commented:
I have converted this PR to a draft. We can revisit this when we are actually in the testing phase of labware 8 and NGS.
